api/v1alpha1/healthcheck_types.go

```diff
@@ -25,11 +25,13 @@ import (
 // Important: Run "make" to regenerate code after modifying this file
 
 // HealthCheckSpec defines the desired state of HealthCheck
+// Either RepeatAfterSec or Scheduler must be defined for the health check to run
 type HealthCheckSpec struct {
-	RepeatAfterSec int      `json:"repeatAfterSec"`
-	Description    string   `json:"description,omitempty"`
-	Workflow       Workflow `json:"workflow"`
-	Level          string   `json:"level,omitempty"` // defines if a workflow runs in a Namespace or Cluster level
+	RepeatAfterSec int           `json:"repeatAfterSec,omitempty"`
+	Description    string        `json:"description,omitempty"`
+	Workflow       Workflow      `json:"workflow"`
+	Level          string        `json:"level,omitempty"`      // defines if a workflow runs in a Namespace or Cluster level
+	Scheduler      SchedulerSpec `json:"scheduler,,omitempty"` // Scheduler defines schedule rules to run HealthCheck
```

Comment:
perhaps an extra comma here:
```suggestion
	Scheduler      SchedulerSpec `json:"scheduler,omitempty"` // Scheduler defines schedule rules to run HealthCheck
```

Reply:
sure updated